### PR TITLE
Intl: Add tests for table iteration order.

### PR DESCRIPTION
**Pull request by Ms2ger**
The order was defined in <https://github.com/tc39/ecma402/pull/279>.

### CODE REVIEW

File: test/intl402/Collator/prototype/resolvedOptions/order.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "collation",
+];
+
+if ("numeric" in options) {
+  expected.push("numeric");
+}
+
+if ("caseFirst" in options) {
+  expected.push("caseFirst");
+}
```
Comment (rwaldron) [inline test/intl402/Collator/prototype/resolvedOptions/order.js]: Based on this: 

> Intl.Collator instances also have the following internal slots if the key corresponding to the name of the internal slot in Table 2 is included in the [[RelevantExtensionKeys]] internal slot of Intl.Collator:
>
> - [[Numeric]] is a Boolean value, specifying whether numeric sorting is used.
> - [[CaseFirst]] is one of the String values "upper", "lower", or "false".


There should be one test file that tests that these properties exist when the conditions are met, and one test file for when the condition is not met.

Comment (Ms2ger) [reply test/intl402/Collator/prototype/resolvedOptions/order.js]: That would be four tests, you mean? How would you distinguish them?

Comment (littledan) [reply test/intl402/Collator/prototype/resolvedOptions/order.js]: Note, the conditions here are choices an implementation can make, not different runtime conditions achievable in a single implementation.

Comment (rwaldron) [reply test/intl402/Collator/prototype/resolvedOptions/order.js]: > the conditions here are choices an implementation can make, 

After further reviewing the section "In Collator:" under [Annex A Implementation Dependent Behaviour
](https://tc39.github.io/ecma402/#annex-implementation-dependent-behaviour), I recognize that this test is valid, however I think a specification that allows implementations to vary in this way is a poor design. All properties listed in each "Resolved Options" table should be present on the object returned by `resolvedOptions()` with a value of `undefined` if no useful value can be determined.

Comment (littledan) [reply test/intl402/Collator/prototype/resolvedOptions/order.js]: I agree that the optionality is a sort of spec smell. I proposed that we make support mandatory in https://github.com/tc39/ecma402/issues/113 but I don't think I did a great job explaining what the change would really mean. Maybe I should try to push for it again.

Comment (littledan) [reply test/intl402/Collator/prototype/resolvedOptions/order.js]: I agree that the optionality is a sort of spec smell. I proposed that we make support mandatory in https://github.com/tc39/ecma402/issues/113 but I don't think I did a great job explaining what the change would really mean. Maybe I should try to push for it again.

**Pull request merged**